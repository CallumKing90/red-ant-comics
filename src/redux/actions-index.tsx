import { actions as applicationStateActions } from "./reducers/applicationState/applicationState";

export default {
  ...applicationStateActions,
};
