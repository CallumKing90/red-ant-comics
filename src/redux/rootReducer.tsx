import { combineReducers } from "@reduxjs/toolkit";
import applicationState from "./reducers/applicationState/applicationState";
import { marvelApi } from "./services/marvel/marvelApi";

export const rootReducer = combineReducers({
  applicationState,
  [marvelApi.reducerPath]: marvelApi.reducer,
});

export default rootReducer;
