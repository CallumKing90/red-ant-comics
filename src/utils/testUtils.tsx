import React from "react";
import { createStore } from "@reduxjs/toolkit";
import { Provider } from "react-redux";
import { render } from "@testing-library/react";

export const renderWithRedux = (
  ui: JSX.Element,
  store = createStore(() => ({}))
) => ({
  ...render(<Provider store={store}>{ui}</Provider>),
  store,
});
