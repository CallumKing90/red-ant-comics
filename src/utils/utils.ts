interface Favourite {
  title: string;
  id: string;
}

export const comicIdInArray = (
  favouritesList: Favourite[],
  comicId: string
) => {
  for (const favourite of favouritesList) {
    if (favourite.id === comicId) return true;
  }

  return false;
};
