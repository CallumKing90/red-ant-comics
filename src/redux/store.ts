import { configureStore } from "@reduxjs/toolkit";
import rootReducer from "./rootReducer";
import { marvelApi } from "./services/marvel/marvelApi";

export const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(marvelApi.middleware),
});

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>;
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch;
