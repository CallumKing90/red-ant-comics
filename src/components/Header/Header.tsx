import React from "react";
import { Button } from "../Button/Button";
import { Heading } from "../Heading/Heading";
import { useAppDispatch } from "../../utils/hooks";
import actions from "../../redux/actions-index";

export function Header(): JSX.Element {
  const dispatch = useAppDispatch();

  const FavouriteClick = () => {
    dispatch(actions.setFavouriteState());
  };

  return (
    <header className="site-header" role="banner">
      <Heading variation="h1" className="site-heading">
        Red Ant Comics
      </Heading>
      <Button
        className="favourites-toggle js-favourites-toggle"
        onClick={FavouriteClick}
      />
    </header>
  );
}
