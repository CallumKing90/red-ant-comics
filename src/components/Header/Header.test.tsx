import { fireEvent, render } from "@testing-library/react";
import React from "react";
import { Header } from "./Header";
import { renderWithRedux } from "../../utils/testUtils";
import { store } from "../../redux/store";
import "@testing-library/jest-dom";
import { createStore } from "@reduxjs/toolkit";
import { debug } from "console";

describe("Header", () => {
  it("should render the header", () => {
    const { getByText, getByRole } = renderWithRedux(<Header />, store);

    const header = getByRole("banner");
    const headerText = getByText("Red Ant Comics");
    expect(header).toBeInTheDocument();
    expect(headerText).toBeInTheDocument();
    expect(header).toMatchSnapshot();
  });

  it("should change the favouritePanelActive state on button click", () => {
    const { getByRole } = renderWithRedux(<Header />, store);

    const favouriteButton = getByRole("button");
    expect(favouriteButton).toHaveClass("favourites-toggle");

    let { favouritesPanelOpen } = store.getState().applicationState;
    expect(favouritesPanelOpen).toBeFalsy();

    fireEvent.click(favouriteButton);

    favouritesPanelOpen = store.getState().applicationState.favouritesPanelOpen;

    expect(favouritesPanelOpen).toBeTruthy();
  });
});
