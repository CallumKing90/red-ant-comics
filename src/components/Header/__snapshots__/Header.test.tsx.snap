// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render the header 1`] = `
<header
  class="site-header"
  role="banner"
>
  <h1
    class="site-heading"
  >
    Red Ant Comics
  </h1>
  <button
    class="favourites-toggle js-favourites-toggle"
    role="button"
  />
</header>
`;
